### PR TITLE
Databases system

### PR DESCRIPTION
**Pull request by ProgrammingWizzard**

### CODE REVIEW

File: src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        return new SQLiteDatabase(file);
+    }
+
+    public static Database createMySQL(String host, int port, String database, String user, String password) {
+        return new MySQLDatabase(host, port, database, user, password);
+    }
+
+    protected abstract Connection getConnection() throws SQLException;
+
+    private final ExecutorService service = Executors.newFixedThreadPool(3);
```
Comment (TheMolkaPL) [inline src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java]: `3` should be configurable

File: src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    public void update(String query, Callback callback) {
+        this.service.submit(() -> {
+            try {
+                Connection connection = this.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query);
```
Comment (TheMolkaPL) [inline src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java]: Your should prepare your query.

File: src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                statement.executeUpdate();
+                if (callback != null) {
+                    callback.onSuccess(null);
+                }
+                statement.close();
+                connection.close();
```
Comment (TheMolkaPL) [inline src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java]: Use try-with-resource

```java
try (Connection connection = this.getConnection()) {
    try (PreparedStatement statement = connection.prepareStatement()) {
        // your code goes here...
    }
}
```

File: src/main/java/com/programmingwizzard/magicguilds/api/database/SQLiteDatabase.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+import com.zaxxer.hikari.HikariDataSource;
+
+import java.io.File;
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public class SQLiteDatabase extends Database {
+
+    private final HikariDataSource source = new HikariDataSource();
```
Comment (TheMolkaPL) [inline src/main/java/com/programmingwizzard/magicguilds/api/database/SQLiteDatabase.java]: You could add something like `HikariDatabase` with this variable.

File: src/main/java/com/programmingwizzard/magicguilds/api/database/SQLiteDatabase.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    public SQLiteDatabase(File file) {
+        if (!file.exists()) {
+            throw new UnsupportedOperationException("File must be a exists!");
+        }
+        this.source.setDataSourceClassName("org.sqlite.SQLiteDataSource");
+        this.source.setMaximumPoolSize(16);
```
Comment (TheMolkaPL) [inline src/main/java/com/programmingwizzard/magicguilds/api/database/SQLiteDatabase.java]: Should be configurable.

File: src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java (hunk 1 above)
Comment (ProgrammingWizzard) [inline src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java]: Configuration will be later.

File: src/main/java/com/programmingwizzard/magicguilds/api/database/SQLiteDatabase.java (hunk 5 above)
Comment (ProgrammingWizzard) [inline src/main/java/com/programmingwizzard/magicguilds/api/database/SQLiteDatabase.java]: Configuration will be later.

File: src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java (hunk 1 above)
Comment (TheMolkaPL) [inline src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java]: You are never shutting down the `ExecutorService`!

Comment (ProgrammingWizzard) [inline src/main/java/com/programmingwizzard/magicguilds/api/database/Database.java]: Sorry, my fault :V

**Pull request merged**